src/controlCenter/controlCenter.vala

```diff
@@ -207,6 +205,37 @@ namespace SwayNotificationCenter {
             });
         }
 
+        /**
+         * Sets the notification window width and height to the user configured
+         * values, taking into account the monitor width and height.
+         * If the configured values are above the monitor's width/height, they
+         * will be clipped to the monitor dimensions.
+         */
+        public void on_map () {
+            // "Constants"
+            bool anchorTop = ConfigModel.instance.positionY == PositionY.TOP;
+            bool fitToScreen = ConfigModel.instance.fit_to_screen;
+            bool anchorLeft = ConfigModel.instance.positionX == PositionX.LEFT;
+            bool anchorRight = ConfigModel.instance.positionX == PositionX.RIGHT;
+            int configuredWidth = ConfigModel.instance.control_center_width;
+            int configuredHeight = ConfigModel.instance.control_center_height;
+
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.LEFT,
+                                      anchorLeft);
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.RIGHT,
+                                      anchorRight);
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.TOP,
+                                      fitToScreen || anchorTop);
+            GtkLayerShell.set_anchor (this,
+                                      GtkLayerShell.Edge.BOTTOM,
+                                      fitToScreen || !anchorTop);
+            // Always set the size request in all events.
+            this.set_size_request (configuredWidth, configuredHeight);
+        }
```

Comment:
Do we even need this method anymore? Seems like removing this and running `this.set_size_request (configuredWidth, configuredHeight);` in `set_anchor()` reduces the lines by a lot. We can merge this when that's fixed :)

Reply:
Noted. I didn't realize the code duplication at the time.